### PR TITLE
Update examples to GA API

### PR DESCRIPTION
**Pull request by nathanielmanistaatgoogle**
Fixes [issue 4977](https://github.com/grpc/grpc/issues/4977).

### CODE REVIEW

File: examples/python/route_guide/run_codegen.py [hunk 1]
```diff
@@ -28,5 +27,16 @@
 # (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 # OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
-# Runs the protoc with gRPC plugin to generate protocol messages and gRPC stubs.
-python -m grpc.tools.protoc -I../../protos --python_out=. --grpc_python_out=. ../../protos/route_guide.proto
```
Comment (kpayson64) [inline examples/python/route_guide/run_codegen.py]: What is the motivation for this change?  My understanding was that python -m grpc.tools.protoc was the "blessed" way to build protos.  If we change this file, then there several references in the documentation to this file that need to be updated as well.

Comment (nathanielmanistaatgoogle) [inline examples/python/route_guide/run_codegen.py]: The motivation for this change is that the tutorial lends itself better to being used with `virtualenv`. Without this change, the shell script would probably have to be updated to accept as an argument the particular `python` to execute.

@hsaliak and I talked about this; I think he had some other reasons as well.

[This pull request](https://github.com/grpc/grpc.github.io/pull/223) by @LisaFC is one of the documentation updates to which you allude, right?

Comment (kpayson64) [inline examples/python/route_guide/run_codegen.py]: I didn't realize this was part of a larger demo change.  To me, the new quick-start feels like it got more complicated, especially the explicit use of Python2.7, which would imply that grpcio or grpcio-tools doesn't work with Python3, which shouldn't be the case.

I'll defer my opinion, but I still see a reference to 
```python -m grpc.tools.protoc -I../../protos --python_out=. --grpc_python_out=. ../../protos/route_guide.proto```

in the aforementioned PR which is no longer accurate after this change.

Comment (nathanielmanistaatgoogle) [inline examples/python/route_guide/run_codegen.py]: There shouldn't be anything _incorrect_ about `python -m grpc.tools.protoc -I../../protos --python_out=. --grpc_python_out=. ../../protos/route_guide.proto`; right? Just that it's no longer the content of the script that the reader runs to generate their code? [I've left a comment in the doc change](https://github.com/grpc/grpc.github.io/pull/223/files#r71206320).

Comment (soltanmm) [inline examples/python/route_guide/run_codegen.py]: There is nothing *incorrect*, but I think the concern is that we've decided to mix-and-match Shell-anese in our United States of Python-published "Baby's First Services" book. It makes sense to e.g. 1) provide a unified front of, "This is the one way to do it if you're just interested in running," or 2) explicitly state "These are the different ways to do it," or somesuch, with there existing arguments that 1 is superior to 2 (I don't have a strong stance here, though).

Comment (kpayson64) [inline examples/python/route_guide/run_codegen.py]: Recapping an offline conversation:  I'm more accepting of this change because we are committing to having protoc.main as part of a GA API.

I'm still not in love with this as a public API though, it seems awkward to have a python function that takes a series of shell style argument strings.  If we are going to vend a way to generate protos in python code, I would much rather see explicit arguments for each property.

Comment (nathanielmanistaatgoogle) [inline examples/python/route_guide/run_codegen.py]: [Agreed](https://github.com/grpc/grpc/issues/7456). Good thing we don't have to worry about "main" squatting the name we'd most prefer.

**Comment by hsaliak** [comment]
LGTM
